lib/paperclip/color_extractor.rb

```diff
@@ -0,0 +1,186 @@
+# frozen_string_literal: true
+
+require 'mime/types/columnar'
+
+module Paperclip
+  class ColorExtractor < Paperclip::Processor
+    MIN_CONTRAST        = 3.0
+    FREQUENCY_THRESHOLD = 0.01
+
+    def make
+      depth = 8
+
+      # Determine background palette by getting colors close to the image's edge only
+      background_palette = palette_from_histogram(convert(':source -alpha set -gravity Center -region \'75%x75%\' -alpha transparent +region -format %c -colors :quantity -depth :depth histogram:info:', source: File.expand_path(@file.path), quantity: 10, depth: depth), 10)
```

Comment:
For some reason it doesn't work on ImageMagick 7.0.9. This is probably a bug in ImageMagick, but the following does work:

```ruby
       background_palette = palette_from_histogram(convert(':source -alpha set -gravity Center -region \'75%x75%\' -fill None -colorize 100% -alpha transparent +region -format %c -colors :quantity -depth :depth histogram:info:', source: File.expand_path(@file.path), quantity: 10, depth: depth), 10)
```